src/controllers/StudentController.ts

```diff
@@ -2,12 +2,97 @@ import { Student } from "../entities/Student";
 
 import { Request, Response } from "express";
 import { Repository } from "typeorm";
+import { validate } from "class-validator";
 
 export const StudentController = (studentRepository: Repository<Student>) => {
   const getAll = async (req: Request, res: Response): Promise<Response> => {
-    const students = await studentRepository.find();
-    return res.status(200).json(students);
+    try {
+      const students = await studentRepository.find();
+      return res.status(200).json(students);
+    } catch (error) {
+      return res.status(500).json({ error: "Unexpected DB error" });
+    }
   };
 
-  return { getAll };
+  const getById = async (req: Request, res: Response): Promise<Response> => {
+    const studentId = req.params.studentId;
+
+    try {
+      const student = await studentRepository.findOne(studentId);
+
+      if (student) {
+        return res.status(200).json(student);
+      }
+      return res.status(404).json({ error: "Student not found" });
+    } catch (error) {
+      return res.status(500).json({ error: "Unexpected DB error" });
+    }
+  };
+
+  const create = async (req: Request, res: Response): Promise<Response> => {
+    const providedStudent = req.body;
+    const [createdStudent] = studentRepository.create([providedStudent]);
+
+    const errors = await validate(createdStudent);
+
+    if (errors.length > 0) {
+      return res.status(400).json({ error: "Invalid student", errors });
+    }
+
+    try {
+      const { enrollmentId: savedStudentId } = await studentRepository.save(
+        createdStudent
+      );
+      const savedStudent = await studentRepository.findOne(savedStudentId);
+      return res.status(200).json(savedStudent);
+    } catch (error) {
+      return res.status(500).json({ error: "Unexpected DB error" });
+    }
+  };
+
+  const update = async (req: Request, res: Response): Promise<Response> => {
+    const studentId = req.params.studentId;
+    const providedStudent = req.body;
+
+    try {
+      const student = await studentRepository.findOne(studentId);
+
+      if (student) {
+        studentRepository.merge(student, providedStudent);
+
+        const errors = await validate(student);
+
+        if (errors.length > 0) {
+          return res.status(400).json({ error: "Invalid student", errors });
+        }
+
+        const { enrollmentId: savedStudentId } = await studentRepository.save(
+          student
+        );
+        const savedStudent = await studentRepository.findOne(savedStudentId);
+        return res.status(200).json(savedStudent);
+      }
+      return res.status(404).json({ error: "Student not found" });
```

Comment:
Intenta evitar el uso de ifs "lejanos", en este caso no encuentras al student pero el error se responde hasta al final, y adicionalmente el código cuenta con una anidación extra a la lógica, te sugiero hacer lo siguiente:
```suggestion
      if (!student) {
         return res.status(404).json({ error: "Student not found" });
      }
      studentRepository.merge(student, providedStudent);

      const errors = await validate(student);

      if (errors.length > 0) {
        return res.status(400).json({ error: "Invalid student", errors });
      }

      const { enrollmentId: savedStudentId } = await studentRepository.save(
        student
       );
      const savedStudent = await studentRepository.findOne(savedStudentId);
      return res.status(200).json(savedStudent);
      
      
```

Reply:
He realizado el cambio, sin embargo ahora uso `Object.assign` en lugar de `studentRepository.merge`. ¿Crees que haya algún problema con esto? Me parece conceptualmente más sencillo pensar en tal función que en la de merge. Aquí está mi [código actual para la función update](https://github.com/HerCerM/uady-devops-final-project/pull/2/files#diff-55d0022ca43eab43eb5abe6886b18965a085339d35027029b2392af36d2b95bfR63).